csrc/runtime/executor_kernel_arg.cpp

```diff
@@ -46,21 +34,64 @@ PrimDataType getSmallestIndexType(const at::Tensor& tensor) {
 
 } // namespace
 
+void KernelArgumentHolder::push(const std::vector<at::Tensor>& tensors) {
+  for (const auto& tensor : tensors) {
+    arguments_.emplace_back(PolymorphicValue(tensor));
+  }
+}
+
 void KernelArgumentHolder::push(const c10::ArrayRef<c10::IValue>& args) {
-  // Naive I/O setup, I'm ignoring all the potential transformation (i.e. I/O
-  // allocated here from the subgraph could be, and very likely are, different
-  // from I/O expected by the generated CUDA kernel.
   for (const auto& arg : args) {
-    push(PolymorphicValue_functions::IValueToPolymorphicValue(arg));
+    arguments_.emplace_back(
+        PolymorphicValue_functions::IValueToPolymorphicValue(arg));
   }
 }
 
-void KernelArgumentHolder::push(const std::vector<at::Tensor>& tensors) {
-  for (const auto& tensor : tensors) {
-    push(tensor);
+void KernelArgumentHolder::push(const std::vector<c10::IValue>& args) {
+  for (const auto& arg : args) {
+    arguments_.emplace_back(
+        PolymorphicValue_functions::IValueToPolymorphicValue(arg));
   }
 }
 
+void KernelArgumentHolder::push(const at::Tensor& tensor) {
+  arguments_.emplace_back(PolymorphicValue(tensor));
+}
+
+void KernelArgumentHolder::push(const std::optional<at::Tensor>& opt_tensor) {
+  if (opt_tensor.has_value()) {
+    arguments_.emplace_back(PolymorphicValue(opt_tensor.value()));
+  }
```

Comment:
Do we need an `else` branch?
```C++
else {
  arguments_.emplace_back(); // emplace a std::monostate, i.e. null
}
```

Reply:
The expectation is we won't push empty optionals, I added an assert.